src/platforms/common/server/quirk_common.cpp

```diff
@@ -105,3 +105,28 @@ auto mir::graphics::common::get_device_driver(mir::udev::Device const* parent_de
     mir::log_warning("udev device has no parent! Unable to determine driver for quirks.");
     return "<UNKNOWN>";
 }
+
+auto mir::graphics::common::apply_quirk(
+    std::string_view devnode,
+    std::string_view driver,
+    std::unordered_map<std::string, std::string> const& devnode_quirks,
+    std::unordered_map<std::string, std::string> const& driver_quirks,
+    std::string_view message) -> std::string
+{
+    if (auto p = devnode_quirks.find(std::string{devnode}); p != devnode_quirks.end())
+    {
+        mir::log_debug("Quirks(%s): forcing %s implementation", message.data(), p->second.c_str());
```

Comment:
Like TICS says, `string_view::data()` doesn't guarantee null termination: `message` should be passed as a `char const*`